### PR TITLE
new #10088 -- Added share lock for QuerySet

### PR DESCRIPTION
**Pull request by smartofmarked**
support read lock for mysql and pgsql , added select_for_share

### CODE REVIEW

**Comment by kezabelle** [comment]
Hi @smartofmarked, thank you for opening this. I've just updated the ticket ([10088](https://code.djangoproject.com/ticket/10088)) to add the `has patch` flag -- beyond that, please be patient. I'm sure it will get looked at as soon as someone has the bandwidth to do so.

**Comment by smartofmarked** [comment]
[kezabelle](https://github.com/kezabelle) , thanks very much , i am so new,  and english so new to , but i am very interested for this, If I do something wrong, please tell me , I promise I'll try do it

**Comment by auvipy** [comment]
please comment on the related ticket and uncheck patch need improvement to make this into review Q

File: docs/releases/4.2.txt [hunk 1]
```diff
@@ -113,6 +113,12 @@ Minor features
 
 * ...
 
+:mod:`django.db.models.query.QuerySet`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
```
Comment (auvipy) [inline docs/releases/4.2.txt:118]: you probably should add some detail here about the feature

File: docs/topics/db/queries.txt [hunk 2]
```diff
@@ -1776,3 +1776,14 @@ Finally, it's important to note that the Django database layer is merely an
 interface to your database. You can access your database via other tools,
 programming languages or database frameworks; there's nothing Django-specific
 about your database.
+
+
+select_for_share
```
Comment (auvipy) [inline docs/topics/db/queries.txt:1781]: this might need version added 4.2 annotation

**Review by auvipy — CHANGES_REQUESTED** [review]

**Comment by ngnpope** [comment]
See #16381 for a more complete implementation.

**Pull request closed without merging**